src/main/java/seedu/address/model/filter/FilterByDatePredicate.java

```diff
@@ -0,0 +1,101 @@
+package seedu.address.model.filter;
+
+import java.time.LocalDateTime;
+import java.util.stream.Stream;
+
+import seedu.address.model.task.DeadlineTask;
+import seedu.address.model.task.EventTask;
+import seedu.address.model.task.FloatingTask;
+/*
+ * Filters the events and deadlines based on the time when they are scheduled.
+ * The user inputs a start time and end time, and all events and tasks that occur between the time range
+ * specified by the user, are displayed on the user interface.
+ *
+ *
+ */
+public class FilterByDatePredicate implements TaskPredicate{
+	private final LocalDateTime startTime;
+	private final LocalDateTime endTime;
+
+	public FilterByDatePredicate (LocalDateTime startTime, LocalDateTime endTime){
+		assert !startTime.equals(null);
+		assert !endTime.equals(null);
+
+		this.startTime = startTime;
+		this.endTime = endTime;
+	}
+
+	@Override
+	public boolean test (DeadlineTask task){
+		if((greaterThan(task) && lesserThan(task)) | equalToMin(task) | equalToMax(task))
+			return true;
+		else
+			return false;
```

Comment:
Now that I have lots of time to read your PR carefully, I *think* this is correct. But, if you wanted to get this PR merged on the deadline day itself, your code needs to be extremely easy to read so I can verify its correctness. Unfortunately, this code, with all the `greaterThan` and `lesserThan` methods, is obfuscated and hard to read.

Basically, I think what you want is this:
```
due >= start && due <= end
```

Using CS1231, we get:
```
!!(due >= start && due <= end)
!(!(due >= start) || !(due <= end)
!(due < start || due > end)
```

So this will accomplish the same thing with less code, and allows me to verify the correctness much more easily:
```java
@Override
public boolean test(DeadlineTask task) {
    // Task due date must be within startTime and endTime inclusive
    return !(task.getDue().isBefore(startTime) || task.getDue().isAfter(endTime));
}
```
The comment tells me what the code should do, and the next line is short and succinct -- allowing me to easily verify its correctness.

This makes *much* more sense to first time readers compared to having all of these `greaterThan`, `lesserThan` methods.